src/org/thoughtcrime/securesms/RegistrationActivity.java

```diff
@@ -198,33 +199,51 @@ public void onClick(View v) {
       int gcmStatus = GooglePlayServicesUtil.isGooglePlayServicesAvailable(self);
 
       if (gcmStatus != ConnectionResult.SUCCESS) {
-        if (GooglePlayServicesUtil.isUserRecoverableError(gcmStatus)) {
+        if(gcmStatus == ConnectionResult.SERVICE_MISSING && BuildConfig.FORCE_WEBSOCKETS) {
+          AlertDialogWrapper.Builder unsupportedDialog = new AlertDialogWrapper.Builder(self);
+          unsupportedDialog.setTitle(getString(R.string.RegistrationActivity_unsupported));
+          unsupportedDialog.setMessage(getString(R.string.RegistrationActivity_websockets_only_unsupported));
+          unsupportedDialog.setPositiveButton(getString(R.string.RegistrationActivity_I_understand),
+                                              new DialogInterface.OnClickListener() {
+                                                @Override
+                                                public void onClick(DialogInterface dialog, int which) {
+                                                  showDoubleCheckDialog(self,e164number);
+                                                }
+                                              });
+          unsupportedDialog.show();
+        } else if (GooglePlayServicesUtil.isUserRecoverableError(gcmStatus)) {
           GooglePlayServicesUtil.getErrorDialog(gcmStatus, self, 9000).show();
+          return;
         } else {
           Dialogs.showAlertDialog(self, getString(R.string.RegistrationActivity_unsupported),
                                   getString(R.string.RegistrationActivity_sorry_this_device_is_not_supported_for_data_messaging));
+          return;
```

Comment:
I'd say make line 202 an OR but still display the warning here and not return
Otherwise we wouldn't gain anything